### DISCUSSION TITLE
Не отображается выбранный ПВЗ

### DISCUSSION THREAD

**Discussion by olgatrick** [cdek-it/wordpress#24] [Ideas]
Добрый день!

Как сделать так, чтобы в письме которое приходит на emal администратора с уведомлением о новом заказе: были добавлены данные о выбранном пвз?
Клиент при оформлении заказа может выбрать пвз по одному адресу, а в поля вбить свой личный домашний адрес. По итогу когда приходит заказ на почту: мы видим только личный адрес клиента и спосбо доставки "СДЭК ДО ПВЗ", а адреса этого выбранного ПВЗ мы не видим.
Как решить эту проблему?

**Comment by olgatrick** [comment]
Или можно ли сделать так, как в других плагинах по доставке: если клиент выбирает какой-то пвз, то данные в поля для заполгнения "адрес" подтягиваются автоматически в зависимости от адреса пвз??

**Comment by vermorag** [comment]
Добрый день.
>Как сделать так, чтобы в письме которое приходит на emal администратора с уведомлением о новом заказе: были добавлены данные о выбранном пвз?

В настоящий момент - никак. У нас уже есть запрос на доработку с данной функциональностью под номером WP-34.
Статус задач, взятых на доработку, вы можете отслеживать в нашей рассылке обновлений. Для подключения к рассылке, напишите в наш чат технической поддержки https://t.me/cdek_it_bot раздел Интеграция для интернет-магазинов.

>Или можно ли сделать так, как в других плагинах по доставке: если клиент выбирает какой-то пвз, то данные в поля для заполгнения "адрес" подтягиваются автоматически в зависимости от адреса пвз??

В старых версиях плагина так и работало, однако были массовые запросы по отключению данной функциональности, соответственно сейчас только планиируется добавить код ПВЗ в шаблон рассылки от вордпресса.

Обращу внимание, что клиенту итак приходит сообщение от компании СДЭК о регистрации заказа со ссылкой, где можно его отследить и посмотреть ПВЗ выдачи.

**Reply by olgatrick** [reply]
Здравствуйте! Прошло 4 месяца. Когда будет решение данного вопроса? 
Или может быть можно вручную вставить какой-либо код в вукомерс по отбражению в письме на емайл этого второго адреса именно выбранного ПВЗ?

**Reply by Dustor** [reply]
Пример моего кода для admin_new_order.php (я не программист, если что, просто скопипастил код из самого же СДЭК):

namespace Cdek\UI;
use Cdek\CdekApi;
use Cdek\Config;
use Cdek\Helpers\WeightCalc;
use Cdek\MetaKeys;
use Exception;
use WC_Order_Item_Shipping;
try {
$shipping_method = $order->get_shipping_method();
$data = $order->get_data(); // данные заказа
$shipping_methods = $order->get_shipping_methods();
foreach ($shipping_methods as $shipping_method) {
$shipping_method_meta_code = $shipping_method->get_meta('_official_cdek_office_code');
if( !empty($shipping_method_meta_code)) {
$officeAddress = json_decode((new CdekApi)->getOffices(['code' => $shipping_method_meta_code])['body']
, true, 512, JSON_THROW_ON_ERROR);

        if (empty($officeAddress[0]['location']['address'])) {
        } else {
            echo '<p style="font-size:1.7em;line-height:1.4em;"><b>Доставка в пункт СДЭК:</b><br> ' 
                . esc_html__( $data['shipping']['state'], 'woocommerce') . ', ' . esc_html__( $data['shipping']['city'], 'woocommerce')  
                . ', ' . esc_html__( $officeAddress[0]['location']['address'], 'woocommerce') . '</p>';
        }
    }
}
